apiserver/facades/client/application/deploy.go

```diff
@@ -294,21 +293,8 @@ func (api *APIBase) addUnits(
 ) ([]coreunit.Name, error) {
 	units := make([]coreunit.Name, 0, n)
 
-	allSpaces, err := api.networkService.GetAllSpaces(ctx)
-	if err != nil {
-		return nil, err
-	}
-
 	// TODO what do we do if we fail half-way through this process?
```

Comment:
While you're here, do you think it's worth resolving this?

`AddIAASUnits` and `AddCAASUnits` take multi-variate `AddUnitArg` params, so we can construct our args upfront and insert all our units in one transaction